### ISSUE TITLE
invalid byte sequence in UTF-8: upload_to_testflight can't process ipa files larger than 4GB

### ISSUE THREAD

**Issue by chenshifei** [fastlane/fastlane#19692]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Recently I got this `invalid byte sequence in UTF-8` error when uploading my ipa file to testflight. When I traced down the error and examine the output from `unzip -Z -1` I found there's some file name containing characters like `ö` or `å`.

These files have been in the app for quite a long time, and before they can be processed without any problems. I noticed that the last successful build was much smaller (about 2.3G) while now the ipa has grown massively into 6G. Could this be the problem?

##### Command executed

`bundle exec fastlane send_to_testflight "scheme:com.test.app" --verbose`

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
[✔] 🚀
DEBUG [2021-12-10 15:13:29.33]: Checking if there are any plugins that should be loaded...
Successfully loaded Appfile at path '/path/to/Appfile'
- apple_id: 'apple-dev@company.com'
-------
DEBUG [2021-12-10 15:13:29.56]: Using scheme: com.test.app
INFO [2021-12-10 15:13:29.57]: ---------------------------
INFO [2021-12-10 15:13:29.57]: --- Step: opt_out_usage ---
INFO [2021-12-10 15:13:29.57]: ---------------------------
INFO [2021-12-10 15:13:29.57]: Disabled upload of used actions
INFO [2021-12-10 15:13:29.57]: ------------------------------
INFO [2021-12-10 15:13:29.57]: --- Step: default_platform ---
INFO [2021-12-10 15:13:29.57]: ------------------------------
INFO [2021-12-10 15:13:29.57]: Driving the lane 'ios send_to_testflight' 🚀
INFO [2021-12-10 15:13:29.58]: -----------------------------
INFO [2021-12-10 15:13:29.58]: --- Step: setup_circle_ci ---
INFO [2021-12-10 15:13:29.58]: -----------------------------
INFO [2021-12-10 15:13:29.58]: Not running on CI, skipping CI setup
INFO [2021-12-10 15:13:29.58]: ---------------------------------------
INFO [2021-12-10 15:13:29.58]: --- Step: app_store_connect_api_key ---
INFO [2021-12-10 15:13:29.58]: ---------------------------------------
INFO [2021-12-10 15:13:29.62]: ----------------------------------
INFO [2021-12-10 15:13:29.62]: --- Step: upload_to_testflight ---
INFO [2021-12-10 15:13:29.62]: ----------------------------------
INFO [2021-12-10 15:13:29.62]: Creating authorization token for App Store Connect API
DEBUG [2021-12-10 15:13:29.63]: App identifier (com.test.app)
INFO [2021-12-10 15:13:34.63]: Ready to upload new build to TestFlight (App: xxxxxxxx)...
INFO [2021-12-10 15:13:34.73]: ▸ warning [/Users/shifei.chen/Repo/ios-app/artifacts/com.test.app.ipa]:  4294967296 extra bytes at beginning or within zipfile
INFO [2021-12-10 15:13:34.73]: ▸   (attempting to process anyway)
WARN [2021-12-10 15:13:34.73]: Lane Context:
INFO [2021-12-10 15:13:34.73]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios send_to_testflight"}
ERROR [2021-12-10 15:13:34.73]: invalid byte sequence in UTF-8
INFO [2021-12-10 15:13:34.75]: Successfully generated documentation at path '/Users/shifei.chen/Repo/ios-app/fastlane/README.md'

+------+---------------------------+-------------+
|                fastlane summary                |
+------+---------------------------+-------------+
| Step | Action                    | Time (in s) |
+------+---------------------------+-------------+
| 1    | opt_out_usage             | 0           |
| 2    | default_platform          | 0           |
| 3    | setup_circle_ci           | 0           |
| 4    | app_store_connect_api_key | 0           |
| 💥   | upload_to_testflight      | 5           |
+------+---------------------------+-------------+

ERROR [2021-12-10 15:13:34.76]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...
Search query: invalid byte sequence in UTF-8

URL: https://api.github.com/search/issues?q=invalid%20byte%20sequence%20in%20UTF-8+repo:fastlane/fastlane
➡️  Any fastlane command: `gsub': invalid byte sequence in UTF-8 (ArgumentError)
    https://github.com/fastlane/fastlane/issues/18723 [closed] 6 💬
    a week ago

➡️  [pilot] The resource 'betaAppReviewSubmissions' does not allow 'DELETE'
    https://github.com/fastlane/fastlane/issues/18408 [open] 22 💬
    12 hours ago

➡️  windows  bale erro
    https://github.com/fastlane/fastlane/issues/18755 [closed] 4 💬
    28 Jul 2021

and 46 more at: https://github.com/fastlane/fastlane/search?q=invalid%20byte%20sequence%20in%20UTF-8&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
bundler: failed to load command: fastlane (/usr/local/lib/ruby/gems/2.7.0/bin/fastlane)
Traceback (most recent call last):
	47: from /usr/local/opt/ruby@2.7/bin/bundle:23:in `<main>'
	46: from /usr/local/opt/ruby@2.7/bin/bundle:23:in `load'
	45: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/exe/bundle:37:in `<top (required)>'
	44: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/friendly_errors.rb:128:in `with_friendly_errors'
	43: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/exe/bundle:49:in `block in <top (required)>'
	42: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli.rb:24:in `start'
	41: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	40: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli.rb:30:in `dispatch'
	39: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	38: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	37: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	36: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli.rb:474:in `exec'
	35: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli/exec.rb:28:in `run'
	34: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli/exec.rb:63:in `kernel_load'
	33: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli/exec.rb:63:in `load'
	32: from /usr/local/lib/ruby/gems/2.7.0/bin/fastlane:23:in `<top (required)>'
	31: from /usr/local/lib/ruby/gems/2.7.0/bin/fastlane:23:in `load'
	30: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/bin/fastlane:23:in `<top (required)>'
	29: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	28: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	27: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
	26: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	25: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:117:in `run!'
	24: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	23: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	22: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	21: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/commands_generator.rb:109:in `block (2 levels) in run'
	20: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	19: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	18: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	17: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	16: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	15: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	14: from Fastfile:228:in `block (2 levels) in parsing_binding'
	13: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	12: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	11: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	10: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	 9: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	 8: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	 7: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	 6: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/actions/upload_to_testflight.rb:34:in `run'
	 5: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/pilot/lib/pilot/build_manager.rb:27:in `upload'
	 4: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/pilot/lib/pilot/manager.rb:85:in `fetch_app_platform'
	 3: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:32:in `fetch_app_platform'
	 2: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:45:in `fetch_info_plist_file'
	 1: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:79:in `fetch_info_plist_with_unzip'
/usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:79:in `split': invalid byte sequence in UTF-8 (ArgumentError)
	47: from /usr/local/opt/ruby@2.7/bin/bundle:23:in `<main>'
	46: from /usr/local/opt/ruby@2.7/bin/bundle:23:in `load'
	45: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/exe/bundle:37:in `<top (required)>'
	44: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/friendly_errors.rb:128:in `with_friendly_errors'
	43: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/gems/2.7.0/gems/bundler-2.2.22/exe/bundle:49:in `block in <top (required)>'
	42: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli.rb:24:in `start'
	41: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	40: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli.rb:30:in `dispatch'
	39: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	38: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	37: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	36: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli.rb:474:in `exec'
	35: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli/exec.rb:28:in `run'
	34: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli/exec.rb:63:in `kernel_load'
	33: from /usr/local/Cellar/ruby@2.7/2.7.5/lib/ruby/2.7.0/bundler/cli/exec.rb:63:in `load'
	32: from /usr/local/lib/ruby/gems/2.7.0/bin/fastlane:23:in `<top (required)>'
	31: from /usr/local/lib/ruby/gems/2.7.0/bin/fastlane:23:in `load'
	30: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/bin/fastlane:23:in `<top (required)>'
	29: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	28: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	27: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
	26: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	25: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:117:in `run!'
	24: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	23: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	22: from /usr/local/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	21: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/commands_generator.rb:109:in `block (2 levels) in run'
	20: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	19: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	18: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	17: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	16: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	15: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	14: from Fastfile:228:in `block (2 levels) in parsing_binding'
	13: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	12: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	11: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	10: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	 9: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	 8: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	 7: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	 6: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane/lib/fastlane/actions/upload_to_testflight.rb:34:in `run'
	 5: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/pilot/lib/pilot/build_manager.rb:27:in `upload'
	 4: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/pilot/lib/pilot/manager.rb:85:in `fetch_app_platform'
	 3: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:32:in `fetch_app_platform'
	 2: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:45:in `fetch_info_plist_file'
	 1: from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:79:in `fetch_info_plist_with_unzip'
/usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.198.1/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:79:in `split': \e[31m[!] invalid byte sequence in UTF-8\e[0m (ArgumentError)
  </pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 11.6                                        |
| Ruby                        | 2.7.5                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.34.1                          |
| Installation Source         | /usr/local/lib/ruby/gems/2.7.0/bin/fastlane |
| Host                        | macOS 11.6 (20G165)                         |
| Ruby Lib Dir                | /usr/local/Cellar/ruby@2.7/2.7.5/lib        |
| OpenSSL Version             | OpenSSL 1.1.1l  24 Aug 2021                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 13.1                                        |
| Swift Version               | 5.5.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
opt_out_usage
default_platform(:ios)

CERT_REPO                     = "git@github.com:company/some-repo-holding-certificates"
TEAM_ID              = "some team id"
TESTFLIGHT_TEAM_ID   = "some team id 2"
ASC_API_ISSUER_ID    = "some issuer id"
ASC_API_KEY_ID       = "some key id"
WORKSPACE_DIR                 = "./ios-app.xcworkspace"
ARTIFACTS_DIR                 = "./artifacts"
DESTINATION                   = "generic/platform=iOS"

platform :ios do
  before_all do
    setup_circle_ci
  end

  desc "TestFlight Distribution" 
  lane :send_to_testflight do |values|
    # Arguments
    scheme                = values[:scheme]

    # Variables
    is_dev =              scheme.start_with?("dev.")
    ipa_file              = "#{ARTIFACTS_DIR}/#{scheme}.ipa"
    testflight_team_id    = TESTFLIGHT_TEAM_ID

    # Prepare App Store Connect API Key
    app_store_connect_api_key(
      key_id: ASC_API_KEY_ID,
      issuer_id: ASC_API_ISSUER_ID,
      key_content: "#{ENV["ASC_API_KEY"]}",
      # Use base64 encoded cert content to avoid newline character issues
      # See https://github.com/fastlane/fastlane/issues/17340
      is_key_content_base64: true
    )

    # Distribute to TestFlight
    if (!is_dev) then
      upload_to_testflight(
        app_identifier:                     scheme,
        ipa:                                ipa_file,
        team_id:                            testflight_team_id,
        skip_waiting_for_build_processing:  true,
        beta_app_feedback_email: "email@company.com",
        beta_app_description: "iOS app",
        demo_account_required: true,
        distribute_external: true,
        notify_external_testers: false,
        beta_app_review_info: {
          contact_email: "email@company.com",
          contact_first_name: "Foo",
          contact_last_name: "Bar,
          contact_phone: "+123445789",
          demo_account_name: "email@company.com",
          demo_account_password: "password",
          notes: "some notes"
        },
        localized_app_info: {
          "default": {
            feedback_email: "email@company.com",
            marketing_url: "https://www.company.com",
            privacy_policy_url: "https://www.company.com",
            description: "iOS app",
          }
        },
        localized_build_info: {
          "default": {
            whats_new: "We improved the app increasing the stability and fixing bugs.",
          }
        }
      )
    end
  end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
apple_id("apple-dev@company.com") # Your Apple email address

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.198.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.4.0        |
| bundler                         | 2.2.22       |
| uri                             | 0.10.0       |
| rake                            | 13.0.6       |
| rexml                           | 3.2.5        |
| CFPropertyList                  | 3.0.5        |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.8.0        |
| artifactory                     | 3.0.15       |
| atomos                          | 0.1.3        |
| aws-eventstream                 | 1.2.0        |
| aws-partitions                  | 1.540.0      |
| aws-sigv4                       | 1.4.0        |
| jmespath                        | 1.4.0        |
| aws-sdk-core                    | 3.124.0      |
| aws-sdk-kms                     | 1.52.0       |
| aws-sdk-s3                      | 1.109.0      |
| babosa                          | 1.0.4        |
| claide                          | 1.0.3        |
| colored                         | 1.2          |
| colored2                        | 3.1.2        |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| declarative                     | 0.0.20       |
| digest-crc                      | 0.6.4        |
| unf_ext                         | 0.0.8        |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| dotenv                          | 2.7.6        |
| emoji_regex                     | 3.2.3        |
| excon                           | 0.89.0       |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| multipart-post                  | 2.0.0        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.8.0        |
| http-cookie                     | 1.0.4        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.2.0        |
| fastimage                       | 2.2.5        |
| gh_inspector                    | 1.1.3        |
| jwt                             | 2.3.0        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| signet                          | 0.16.0       |
| googleauth                      | 1.1.0        |
| httpclient                      | 2.8.3        |
| mini_mime                       | 1.1.2        |
| trailblazer-option              | 0.1.2        |
| uber                            | 0.1.0        |
| representable                   | 3.1.1        |
| retriable                       | 3.1.2        |
| webrick                         | 1.7.0        |
| google-apis-core                | 0.4.1        |
| google-apis-androidpublisher_v3 | 0.14.0       |
| google-apis-playcustomapp_v1    | 0.6.0        |
| google-apis-iamcredentials_v1   | 0.8.0        |
| google-apis-storage_v1          | 0.10.0       |
| google-cloud-env                | 1.5.0        |
| google-cloud-errors             | 1.2.0        |
| google-cloud-core               | 1.6.0        |
| google-cloud-storage            | 1.35.0       |
| json                            | 2.6.1        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| optparse                        | 0.1.1        |
| plist                           | 3.6.0        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.3        |
| simctl                          | 1.6.8        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.8.0        |
| terminal-table                  | 1.8.0        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.21.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
</details>

*generated on:* **2021-12-10**
</details>

**Comment by chenshifei** [comment]
I believe this issue is connected to the PR #17825 which introduced `unzip`. However it seems like `unzip` has problems with encodes other than UTF-8, see discussions about this [here](https://superuser.com/questions/872596/decompress-zip-with-given-encoding)

So perhaps you have to choose between these two things: either only use UTF-8 characters in your file names while making massive IPAs, or pick whatever file names but keep the final IPA small.

**Comment by chenshifei** [comment]
Closed since the PR is merged in version 2.200.0.
